### PR TITLE
JSON-LD term drop

### PR DESCRIPTION
**Pull request by aljones15**
> Implementations that use JSON-LD processing, such as RDF Dataset Canonicalization [[RDF-CANON](https://w3c.github.io/vc-data-integrity/#bib-rdf-canon)], MUST throw an error, which SHOULD be DATA_LOSS_DETECTION_ERROR, when data is dropped by a JSON-LD processor, such as when an undefined term is detected in an input document.

https://w3c.github.io/vc-data-integrity/#securing-data-losslessly:~:text=Implementations%20that%20use%20JSON%2DLD%20processing%2C%20such%20as%20RDF%20Dataset%20Canonicalization%20%5BRDF%2DCANON%5D%2C%20MUST%20throw%20an%20error%2C%20which%20SHOULD%20be%20DATA_LOSS_DETECTION_ERROR%2C%20when%20data%20is%20dropped%20by%20a%20JSON%2DLD%20processor%2C%20such%20as%20when%20an%20undefined%20term%20is%20detected%20in%20an%20input%20document.

### CODE REVIEW

**Comment by BigBlueHat** [comment]
@aljones15 a few lint issues that'll block this merge need addressing.

**Review by dlongley — APPROVED** [review]
AFAICT, this makes sense at a high-/concept-level, needs review from others for the details related to test suite-specific implementation.

**Review by BigBlueHat — CHANGES_REQUESTED** [review]

File: assertions.js
Comment (aljones15) [reply assertions.js]: I changed it to `shouldFailIssuance` here: https://github.com/w3c-ccg/data-integrity-test-suite-assertion/pull/90/commits/32294d7a1483cf86b5d28ad07fb8d6e5b6520edb I can also change to `shouldFailVerification` in a later PR. This PR is a bit packed right now.

**Comment by aljones15** [comment]
p.s. the X for node 20 is because the now optional context injection tests are failing because this PR provides a somewhat real issuer. Can be fixed here or later. As this PR is 15 commits I am for later.

**Review by PatStLouis — APPROVED** [review]

**Review by BigBlueHat — APPROVED** [review]

**Pull request merged**
